### ISSUE TITLE
dzvents method silent() no longer work for sensor type devices   

### ISSUE THREAD

**Issue by rwaaren** [domoticz/domoticz#3626]
After removing the updateDevice function in EventSystem.cpp and call the updateDevice function in MainWorker.cpp, ( Somewhere from Beta >=  V4.11305 )the option silent() in dzVents (= NOTRIGGER  in Lua) does not seem to have the desired effect that no event trigger is issued when the device is updated.

Please advise on how to restore that functionality

**Comment by gizmocuz** [comment]
@rwaaren , i need a script example to be able to replicate this

**Comment by rwaaren** [comment]
test Script
```
--[[

V4.11298 When using method silent() ==>> No sensors / switches trigger the script.
V4.11305 When using method silent() ==>> All sensors trigger the script. Switch does not
V4.11316 When using method silent() ==>> All sensors trigger the script. Switch does not

--]]
return 
{
	on = 
	{
		devices = { 'vd*' },
		timer = {'at 09:*'},
	},
	execute = function(dz, item)
		if item.isTimer then
            dz.devices("vdGas").updateGas(10).silent().afterSec(07)
            dz.devices("vdHumidity").updateHumidity(10, 3).silent().afterSec(08)
            dz.devices("vdLeafWetness").updateWetness(10).silent().afterSec(09)
            dz.devices("vdLux").updateLux(10).silent().afterSec(10)
            dz.devices("vdP1SmartMeterElectric").updateP1(10,10,10,10,10,10).silent().afterSec(11)
            dz.devices("vdPercentage").updatePercentage(10).silent().afterSec(12)
            dz.devices("vdPressureBar").updatePressure(10).silent().afterSec(13)
            dz.devices("vdRain").updateRain(10,10).silent().afterSec(14)
            dz.devices("vdScaleWeight").updateWeight(10).silent().afterSec(15)
            dz.devices("vdSoilMoisture").updateSoilMoisture(10).silent().afterSec(16)
            dz.devices("vdSolarRadiation").updateRadiation(10).silent().afterSec(17)
            dz.devices("vdSoundLevel").updateSoundLevel(10).silent().afterSec(18)
            dz.devices("vdTemperature").updateTemperature(10).silent().afterSec(19)
            dz.devices("vdTempHum").updateTempHum(10,3,3).silent().afterSec(20)        
 dz.devices("vdTempHumBaro").updateTempHumBaro(10,1000,'thunderstorm').silent().afterSec(21)
            dz.devices("vdText").updateText('10').silent().afterSec(22)
            dz.devices("vdUsageElectric").updateEnergy(10).silent().afterSec(24)
            dz.devices("vdUV").updateUV(10).silent().afterSec(25)
            dz.devices("vdVisibility").updateVisibility(10).silent().afterSec(26)
            dz.devices("vdVoltage").updateVoltage(10).silent().afterSec(27)
            dz.devices("vdWaterflow").updateWaterflow(10).silent().afterSec(28)
            dz.devices("vdWind").updateWind(10,'N',10,10,10,10).silent().afterSec(29)
            dz.devices("vdTempBaro").updateTempBaro(10,1000,'thunderstorm').silent().afterSec(31)
            dz.devices("vdSwitch").switchOn().silent().afterSec(32)
       	else
		    dz.log('Sensor ' .. item.name .. ' triggered this script', dz.LOG_ERROR)
		end
	end
}
```

Test database:

[domoticz.zip](https://github.com/domoticz/domoticz/files/3613507/domoticz.zip)

**Comment by rwaaren** [comment]
@gizmocuz 
Tested and seem solved. Thx !!  
with commit  [Lua UpdateDevice now signals pushers (WebSocket)](https://github.com/domoticz/domoticz/commit/0697504ea726663306d30bd120913275528f459c) ?

**Comment by gizmocuz** [comment]
@rwaaren , I did not test this yet, but it could be solved. (Lua/EventSystem)
There is now a global boolean (g_bUseEventTrigger) that is set in CMainWorker::UpdateDevice, this can be used (and is now used in EventSystem.cpp), but might also need to be implemented in the Python plugin system

'silent' , it still confuses me... does this mean "no log line", or do not cause a trigger to be started ?
If it is the last one, then it is indeed now solved

**Comment by rwaaren** [comment]
It indeed means not causing a trigger to be started. and I can confirm it is solved in V4.11324 

Thx again !
